synapse/membrane.py

```diff
@@ -0,0 +1,169 @@
+# FIXME support ival splices
+# FIXME implement roles/groups/users/perms
+
+import copy
+import fnmatch
+import operator
+
+from synapse.eventbus import EventBus
+import synapse.lib.syntax as s_syntax
+
+class Membrane(EventBus):
+
+    def __init__(self, src=None, dst=None, rules=None, default=None):
+        '''
+        Filters Cortex splices based on rules and forwards to destination Cortex if rules allow.
+
+        Args:
+            dst: the destination Cortex to forward messages to
+            rules: the filter rules
+        '''
+        EventBus.__init__(self)
+
+        _SPLICE_NAMES = (
+            'node:add',
+            'node:del',
+            'node:prop:set',
+            'node:prop:del',
+            'node:tag:add',
+            'node:tag:del',
+            #'node:ival:set',
```

Comment:
not implemented yet. I want to get the matching / forwarding stuff settled first.